cfpb/crawl-cfgov
# Remove CSS references from archived HTML + sed script reorg

This change adds a new postprocessing sed script that removes CSS references from HTML before it gets committed back to the repository. This fixes #25.

This also refactors how the sed scripts are organized to make them easier to maintain/add and adds a little testing script to validate their behavior, which is also added as a new GitHub workflow on PRs/commits. I also added a little bit of documentation around these scripts.

@schbetsy I've run this script on [my fork](https://github.com/chosak/crawl-cfgov/) with a crawl of depth=1. See large commit [here](https://github.com/chosak/crawl-cfgov/commit/442f6dbaced8eeae57dedf2bd3fa1eb60472ff6c) -- with nicer commit message!

To test, run:

```
postprocessing/transform_results.sh www.consumerfinance.gov
```

This will apply all of the sed scripts on your local archive. I notice that this removes some empty lines that for some reason weren't being removed previously ([example](https://github.com/cfpb/crawl-cfgov/blob/84c7764bfcee5770fc193ae3ba5f93a6ae723a59/www.consumerfinance.gov/consumer-tools/money-as-you-grow/bookshelf/index.html#L1453)):

```diff
diff --git a/www.consumerfinance.gov/consumer-tools/money-as-you-grow/bookshelf/index.html b/www.consumerfinance.gov/consumer-tools/money-as-you-grow/bookshelf/index.html
index 2947b908ba..b6e9c5ae58 100644
--- a/www.consumerfinance.gov/consumer-tools/money-as-you-grow/bookshelf/index.html
+++ b/www.consumerfinance.gov/consumer-tools/money-as-you-grow/bookshelf/index.html
@@ -23,9 +23,8 @@
     <meta property="og:site_name" content="Consumer Financial Protection Bureau">
     <meta property="fb:app_id" content="210516218981921">
     <link rel="shortcut icon" type="image/x-icon" href="/static/favicon.5df6b671015e.ico">
-    
-    <link rel="stylesheet" href="/static/css/main.4e60cb13d1fb.css">
-    <link rel="preload" href="/static/css/main.4e60cb13d1fb.css" as="style">
+    
+    
     <link rel="preload"
           href="/static/fonts/2cd55546-ec00-4af9-aeca-4a3cd186da53.457e94a5b665.woff2"
           as="font"
@@ -1450,7 +1449,6 @@
     <thead>
         <tr>
             <th scope="col" class="u-w20pct">
-     
             </th>
             <th scope="col" class="u-w50pct">
                     <button class="sortable sorted-up"
```

I'm really not sure how or why these escaped the existing script.
